modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/scanner/InboxScannerService.java

```diff
@@ -172,6 +187,20 @@ public synchronized void updated(Dictionary properties) throws ConfigurationExce
     final int maxTries = NumberUtils.toInt(Objects.toString(properties.get(INBOX_TRIES), "3"));
     final int secondsBetweenTries = NumberUtils.toInt(Objects.toString(properties.get(INBOX_TRIES_BETWEEN_SEC), "300"));
 
+    // Metadata parsing configuration
+    var metadataPattern = Optional.ofNullable(properties.get(INBOX_METADATA_REGEX))
+            .map(Objects::toString)
+            .map(Pattern::compile);
+    var dateFormatter = Optional.ofNullable(properties.get(INBOX_DATETIME_FORMAT))
+            .map(Objects::toString)
+            .map(DateTimeFormatter::ofPattern)
+            .orElse(DateTimeFormatter.ISO_DATE_TIME);
+    var ffprobe = BooleanUtils.toBoolean((String) properties.get(INBOX_METADATA_FFPROBE))
```

Comment:
Not sure what you mean. The result is always a `String` and always used as such. Only the first argument of the ternary operator is a boolean. This is just saying, if using ffprobe is turned on, set this to the configured ffprobe binary, otherwise set this to `null`.

I guess you could treat both configuration options differently here, but then you would have a similar check at the end.